jetty-websocket/websocket-javax-common/src/main/java/org/eclipse/jetty/websocket/javax/common/JavaxWebSocketFrameHandler.java

```diff
@@ -278,24 +282,41 @@ public void onFrame(Frame frame, Callback callback)
             dataType = OpCode.UNDEFINED;
     }
 
+    public void onClose(Frame frame, Callback callback)
+    {
+        notifyOnClose(CloseStatus.getCloseStatus(frame), callback);
+    }
+
     @Override
     public void onClosed(CloseStatus closeStatus, Callback callback)
     {
+        notifyOnClose(closeStatus, callback);
+        container.notifySessionListeners((listener) -> listener.onJavaxWebSocketSessionClosed(session));
```

Comment:
`notifyOnClose()` has a guard to avoid notifying more than once, but then `container.notifySessionListeners()` will be called more than once, so either `notifyOnClose()` returns a `boolean` or you need to be tighter with the once-only semantic.

Reply:
This method is for the websocket-core `onClosed` event which is only ever called once. We also call `notifyOnClose()` when we receive a close frame which is why it needs the guard and `container.notifySessionListeners()` does not.